internal/pkg/cli/env_init.go

```diff
@@ -257,10 +261,21 @@ func (o *initEnvOpts) Ask() error {
 
 // Execute deploys a new environment with CloudFormation and adds it to SSM.
 func (o *initEnvOpts) Execute() error {
-	o.initRuntimeClients()
+	if err := o.initRuntimeClients(); err != nil {
+		return err
+	}
+	appVersion, err := o.appVersionGetter.Version()
+	if err != nil {
+		return fmt.Errorf("get template version of application %s: %w", o.appName, err)
+	}
+	if diff := semver.Compare(appVersion, version.LatestTemplateVersion()); diff > 0 && !o.allowAppDowngrade {
```

Comment:
Can you help me understand why we need to check the app version while running `env init`? Is it because of the stack set?

Reply:
Yeah we'll update the stackset, which could cause version downgrade issue.